server/src/main/java/org/opensearch/rest/RequestLimitSettings.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest;
+
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.routing.IndexRoutingTable;
+import org.opensearch.cluster.routing.IndexShardRoutingTable;
+import org.opensearch.cluster.routing.RoutingTable;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.rest.action.cat.RestIndicesAction;
+import org.opensearch.rest.action.cat.RestSegmentsAction;
+import org.opensearch.rest.action.cat.RestShardsAction;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+/**
+ * Class to define dynamic settings for putting circuit breakers on the actions and functions to evaluate if block is required.
+ */
+public class RequestLimitSettings {
+
+    /**
+     * Enum to represent action names against whom we need to perform limit checks.
+     */
+    public enum BlockAction {
```

Comment:
I think this should be named ```BlockingEntity``` and the RestActions can provide clusterState, the blocking entity and their respective limit for the entity.

```

public enum BlockAction {
    INDICES,
    SHARDS
}
```

This would help us with cases where an action needs to block requests based on an entity. For instance, CatSegments would need to block a request based on say SHARDS, because number of segments can not be known upfront from clusterState.